### PR TITLE
Expanded contribution docs

### LINKED ISSUE #716: Remove the Python3.5 Trove Classifier
We actually only support Python 3.6+ | see https://github.com/nucypher/nucypher/blob/master/setup.py#L148

### PR DESCRIPTION
**Pull request by KPrasch**
- Explains the general preferred development submission process in documentation
- Fixes #716

### CODE REVIEW

File: docs/source/guides/contribution_guide.rst [hunk 1]
```diff
@@ -5,6 +5,53 @@ Contributing
...
+----------------------
+
+.. _`NuCypher GitHub`: https://github.com/nucypher/nucypher
+
+In order to contribute new code or documentation changes, you will need a local copy
+of the source code which is located on the `NuCypher GitHub`_.
+
+.. note::
+
+   NuCypher uses ``git`` for version control; Be sure you have it installed.
```
Comment (mswilkison) [inline docs/source/guides/contribution_guide.rst]:
```suggestion
   NuCypher uses ``git`` for version control. Be sure you have it installed.
```

File: docs/source/guides/contribution_guide.rst [hunk 2]
```diff
@@ -5,6 +5,53 @@ Contributing
...
+
+Here is the recommended procedure for acquiring the code in preparation for
+contributing proposed changes:
+
+
+1. Use GitHub to Fork the `nucypher/nucypher` repository
```
Comment (mswilkison) [inline docs/source/guides/contribution_guide.rst]:
```suggestion
1. Use GitHub to Fork the ``nucypher/nucypher`` repository
```

File: docs/source/guides/contribution_guide.rst [hunk 3]
```diff
@@ -5,6 +5,53 @@ Contributing
...
+
+   $ git clone https://github.com/<YOUR-GITHUB-USERNAME>/nucypher.git
+
+3. Change Directories into ``nucypher``
+
+.. code-block:: bash
+
+   cd nucypher
+
+3. Add `nucypher/nucypher` as an upstream remote
```
Comment (mswilkison) [inline docs/source/guides/contribution_guide.rst]:
```suggestion
3. Add ``nucypher/nucypher`` as an upstream remote
```

File: docs/source/guides/contribution_guide.rst [hunk 4]
```diff
@@ -33,6 +80,92 @@ Optionally, to run the full, slow, verbose test suite run:
 
   (nucypher)$ pytest --runslow -s
 
+
+Making A Commit
+---------------
+
+NuCypher takes pride in it's commit history.
```
Comment (mswilkison) [inline docs/source/guides/contribution_guide.rst]:
```suggestion
NuCypher takes pride in its commit history.
```

File: docs/source/guides/contribution_guide.rst [hunk 5]
```diff
@@ -33,6 +80,92 @@ Optionally, to run the full, slow, verbose test suite run:
...
+
+We understand that different code authors have different development preferences, and others
+are first-time contributors to open source, so feel free to join our discord and let us know
+how we can best support the submission of your proposed changes.
+
+
+Opening A Pull Request
+----------------------
+
+When considering including commits as part of a pull request into `nucypher/nucypher`,
```
Comment (mswilkison) [inline docs/source/guides/contribution_guide.rst]:
```suggestion
When considering including commits as part of a pull request into ``nucypher/nucypher``,
```

File: docs/source/guides/contribution_guide.rst [hunk 6]
```diff
@@ -33,6 +80,92 @@ Optionally, to run the full, slow, verbose test suite run:
...
+we *highly* recommend opening the pull request early, before it is finished with
+the mark "[WIP]" prepended to the title.  We understand PRs marked "WIP" to be subject to change,
+history rewrites, and CI failures. Generally we will not review a WIP PR until the "[WIP]" marker
+has been removed from the PR title - However this does give other contributors an opportunity
```
Comment (mswilkison) [inline docs/source/guides/contribution_guide.rst]:
```suggestion
has been removed from the PR title, however, this does give other contributors an opportunity
```

File: docs/source/guides/contribution_guide.rst [hunk 7]
```diff
@@ -33,6 +80,92 @@ Optionally, to run the full, slow, verbose test suite run:
...
+   Be certain you do not have uncommitted changes before continuing.
+
+1. Update your remote tracking branches
+
+.. code-block:: bash
+
+   $ git remote update
+   ...  (some upstream changes are reported)
+
+2. Initiate an interactive rebase over `nucypher/nucypher@master`
```
Comment (mswilkison) [inline docs/source/guides/contribution_guide.rst]:
```suggestion
2. Initiate an interactive rebase over ``nucypher/nucypher@master``
```

File: docs/source/guides/installation_guide.md [hunk 8]
```diff
@@ -1,6 +1,14 @@
 # Installation Guide
 
 
+## Contents
+
+* [Running Ursula with Systemd] (#System-Requirements)
```
Comment (mswilkison) [inline docs/source/guides/installation_guide.md]:
```suggestion
* [Running Ursula with Systemd](#System-Requirements)
```

File: docs/source/guides/installation_guide.md [hunk 9]
```diff
@@ -1,6 +1,14 @@
...
+* [Running Ursula with Systemd] (#Systemd-Service-Installation)
```
Comment (mswilkison) [inline docs/source/guides/installation_guide.md]:
```suggestion
* [Running Ursula with Systemd](#Systemd-Service-Installation)
```

File: docs/source/guides/installation_guide.md [hunk 10]
```diff
@@ -1,6 +1,14 @@
...
+* [Standard Installation] (#Standard-Installation)
```
Comment (mswilkison) [inline docs/source/guides/installation_guide.md]:
```suggestion
* [Standard Installation](#Standard-Installation)
```

File: docs/source/guides/installation_guide.md [hunk 11]
```diff
@@ -1,6 +1,14 @@
...
+* [Development Installation] (#Development-Installation)
```
Comment (mswilkison) [inline docs/source/guides/installation_guide.md]:
```suggestion
* [Development Installation](#Development-Installation)
```

File: docs/source/guides/installation_guide.md [hunk 12]
```diff
@@ -1,6 +1,14 @@
...
+* [Running Ursula with Systemd] (#Systemd Service Installation)
```
Comment (mswilkison) [inline docs/source/guides/installation_guide.md]:
```suggestion
* [Running Ursula with Systemd](#Systemd Service Installation)
```

File: docs/source/guides/installation_guide.md [hunk 13]
```diff
@@ -111,9 +119,12 @@ Ensure you have `git` installed ([Git Documentation](https://git-scm.com/doc)).
 
 
 ### Acquire NuCypher Codebase
+
+Fork the nucypher repository on GitHub, as explained in the [Contribution Guide](/guides/contribution_guide),
```
Comment (mswilkison) [inline docs/source/guides/installation_guide.md]:
```suggestion
Fork the ``nucypher`` repository on GitHub, as explained in the [Contribution Guide](/guides/contribution_guide),
```

File: docs/source/guides/contribution_guide.rst (hunk 2 above)
Comment (KPrasch) [reply docs/source/guides/contribution_guide.rst]: I actually intend for these to be rendered Italic as to not be confused with code.

File: docs/source/guides/installation_guide.md (hunk 11 above)
Comment (KPrasch) [reply docs/source/guides/installation_guide.md]: Oh - Is the formatting wrong here? I think better to not make this change as a GH suggestion, Ill make a new commit.

File: docs/source/guides/contribution_guide.rst (hunk 2 above)
Comment (mswilkison) [reply docs/source/guides/contribution_guide.rst]: ok 👍

File: docs/source/guides/installation_guide.md (hunk 11 above)
Comment (mswilkison) [reply docs/source/guides/installation_guide.md]: For me, it only displays as a URL with no space.

File: setup.py [hunk 14]
```diff
@@ -145,7 +145,6 @@ def run(self):
           "Operating System :: OS Independent",
           "Programming Language :: Python",
           "Programming Language :: Python :: 3 :: Only",
-          "Programming Language :: Python :: 3.5",
```
Comment (KPrasch) [inline setup.py]: Fixed #716

File: docs/source/guides/contribution_guide.rst (hunk 2 above)
Comment (KPrasch) [reply docs/source/guides/contribution_guide.rst]: :heavy_check_mark:

File: docs/source/guides/contribution_guide.rst (hunk 3 above)
Comment (KPrasch) [reply docs/source/guides/contribution_guide.rst]: :heavy_check_mark:

File: docs/source/guides/contribution_guide.rst (hunk 5 above)
Comment (KPrasch) [reply docs/source/guides/contribution_guide.rst]: :heavy_check_mark:

File: docs/source/guides/contribution_guide.rst (hunk 7 above)
Comment (KPrasch) [reply docs/source/guides/contribution_guide.rst]: :heavy_check_mark:

File: docs/source/guides/installation_guide.md (hunk 8 above)
Comment (KPrasch) [reply docs/source/guides/installation_guide.md]: :heavy_check_mark:

File: docs/source/guides/installation_guide.md (hunk 9 above)
Comment (KPrasch) [reply docs/source/guides/installation_guide.md]: :heavy_check_mark:

File: docs/source/guides/installation_guide.md (hunk 10 above)
Comment (KPrasch) [reply docs/source/guides/installation_guide.md]: :heavy_check_mark:

File: docs/source/guides/installation_guide.md (hunk 11 above)
Comment (KPrasch) [reply docs/source/guides/installation_guide.md]: :heavy_check_mark:

File: docs/source/guides/installation_guide.md (hunk 12 above)
Comment (KPrasch) [reply docs/source/guides/installation_guide.md]: :heavy_check_mark:

File: docs/source/guides/installation_guide.md (hunk 13 above)
Comment (KPrasch) [reply docs/source/guides/installation_guide.md]: :heavy_check_mark:

File: docs/source/guides/installation_guide.md (hunk 10 above)
Comment (KPrasch) [reply docs/source/guides/installation_guide.md]: :heavy_check_mark:

**Pull request merged**